packages/wasm/types/index.d.ts

```diff
@@ -9,6 +9,23 @@ import type { FilterPattern } from '@rollup/pluginutils';
  */
 export type TargetEnv = 'auto' | 'auto-inline' | 'browser' | 'node';
 
+/**
+ * The type for the plugin's loader function
+ *
+ * This is the function that ends up called when encountering a WASM import to load it and turn it into an usable object at runtime.
+ *
+ * @param {boolean} sync Whether the load should happen synchronously or not
+ * @param {string | null} filepath The path to the module.
+ * @param {string | null} src The base64-encoded source of the module
+ * @param {any} imports An object containing the module's imports
+ */
+export type WasmLoaderFunction = (
```

Comment:
Possibly? I don't feel confident on that part. It might be possible to hide this type externally, but IMO I'd prefer to be sure we agree on the way forward w.r.t the loader as a function/name issue first.